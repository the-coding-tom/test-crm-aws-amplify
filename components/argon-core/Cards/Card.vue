<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
