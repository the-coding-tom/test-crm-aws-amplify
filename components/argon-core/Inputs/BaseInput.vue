<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
