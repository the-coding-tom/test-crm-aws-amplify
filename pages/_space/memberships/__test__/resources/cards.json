{
  "data": [
    {
      "id": "ccof:Ksaklsalksaklsak",
      "card_brand": "MASTERCARD",
      "last_4": "9399",
      "exp_month": 6,
      "exp_year": 2022,
      "cardholder_name": "Al Ksal"
    }
  ]
}
