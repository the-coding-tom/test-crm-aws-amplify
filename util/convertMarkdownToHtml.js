export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
