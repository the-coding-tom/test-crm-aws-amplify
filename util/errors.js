export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
