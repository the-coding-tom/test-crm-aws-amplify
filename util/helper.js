const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
