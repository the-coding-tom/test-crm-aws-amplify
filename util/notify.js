const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
