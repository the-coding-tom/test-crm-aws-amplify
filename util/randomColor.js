const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
