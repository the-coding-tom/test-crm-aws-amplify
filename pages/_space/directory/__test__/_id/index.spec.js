import IdIndexPage from '@/pages/_space/directory/_id/index.vue'
import MembershipIdIndexPage from '@/pages/_space/memberships/_id/index.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BaseHeader from '~/components/argon-core/BaseHeader'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'
import Card from '~/components/argon-core/Cards/Card.vue'
import Badge from '~/components/argon-core/Badge.vue'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import lodash from 'lodash'
import moment from 'moment'

import allPlans from '@/pages/_space/plans/__test__/resources/all-plans.json'
import allMemberships from '@/pages/_space/memberships/__test__/resources/all-memberships.json'
import singleMembership from '@/pages/_space/memberships/__test__/resources/single-membership.json'
import cards from '@/pages/_space/memberships/__test__/resources/cards.json'

describe('Page: /:space/directory/ | Directory Id Index Page', () => {
  let NuxtStore, store, wrapper, $membership, router, $checkin, $plan

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $membership = {
      getPaymentMethods: jest.fn().mockResolvedValue(cloneDeep(cards)),
      getSubscriptions: jest.fn().mockResolvedValue({ data: [] }),
      getPaidFor: jest.fn().mockResolvedValue({ data: [] }),
      getAllMemberships: jest.fn().mockResolvedValue({
        data: cloneDeep(allMemberships),
        meta: { total: 1, per_page: 15, current_page: 1 },
        links: {}
      }),
      getAMembership: jest
        .fn()
        .mockResolvedValue({ data: cloneDeep(singleMembership) }),
      deleteMembership: jest.fn().mockResolvedValue({ data: {} }),
      addPlan: jest.fn().mockResolvedValue(),
      cancelSubscription: jest.fn().mockResolvedValue({})
    }

    $checkin = {
      checkin: jest.fn().mockResolvedValue(),
      checkout: jest.fn().mockResolvedValue(),
      checkins: jest.fn().mockResolvedValue({
        data: [],
        meta: { total: 1, per_page: 15, current_page: 1 },
        links: {}
      })
    }

    $plan = {
      getAllPlan: jest.fn().mockResolvedValue(cloneDeep(allPlans))
    }

    global._ = lodash
  })

  beforeEach(async done => {
    store = await NuxtStore.createStore()

    router = new VueRouter({
      routes: [
        {
          path: '_space/memberships/_id',
          component: MembershipIdIndexPage,
          name: 'space-memberships-id'
        }
      ]
    })

    const route = {
      query: {
        page: 1
      }
    }

    const params = {
      id: singleMembership.id
    }

    const asyncDataResponse = await IdIndexPage.asyncData({
      store,
      $membership,
      route,
      params
    })

    wrapper = createWrapper(
      IdIndexPage,
      store,
      router,
      {
        plan_id: '',
        loading: false,
        checked: false,
        cards: [],
        currentPage: 1,
        perPage: 5,
        drawer: false,
        direction: 'rtl',
        ...asyncDataResponse
      },
      {
        stubs: {
          'nuxt-link': true,
          'no-ssr': true,
          transition: false,
          'b-popover': true,
          'date-picker': {
            template: '<div>date-picker-stub</div>',
            props: {
              value: ''
            }
          }
        },
        mocks: {
          $membership: $membership,
          $checkin: $checkin,
          $moment: moment,
          $plan: $plan
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(Card.name, Card)
        vueInstance.component(Badge.name, Badge)
      }
    )

    await flushPromises()
    wrapper.vm.$nextTick(done)
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('adds a plan', async () => {
    wrapper.find('#addPlanBtn').trigger('click')
    await wrapper.vm.$nextTick()

    //  select plan
    const planOptions = wrapper.vm.$el.querySelectorAll(
      '#selectPlan .el-select-dropdown__item'
    )
    const evt = document.createEvent('MouseEvents')
    evt.initEvent('mouseenter')
    planOptions[1].dispatchEvent(evt)
    planOptions[1].click()
    await flushPromises()

    // select member
    const cardOptions = wrapper.vm.$el.querySelectorAll(
      '#memberCard .el-select-dropdown__item'
    )
    const evtC = document.createEvent('MouseEvents')
    evtC.initEvent('mouseenter')
    cardOptions[0].dispatchEvent(evtC)
    cardOptions[0].click()
    await flushPromises()

    // mock location.reload()
    const { location } = window
    delete window.location
    window.location = { ...location, reload: jest.fn() }

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($membership.addPlan).toHaveBeenCalled()
    expect($membership.addPlan.mock.calls[0][1].plan_id).toBe(
      allPlans.data.data[1].id
    )
    expect($membership.addPlan.mock.calls[0][1].card_nonce).toBe(
      cards.data[0].id
    )
  })

  it('cancel a plan', async () => {
    // mock confirm
    global.confirm = jest.fn().mockReturnValue(true)

    wrapper
      .find('button.btn.text-danger.btn-transparent.btn-sm')
      .trigger('click')

    expect($membership.cancelSubscription).toHaveBeenCalled()
  })

  it('deletes a member', async () => {
    // mock confirm
    global.confirm = jest.fn().mockReturnValue(true)

    wrapper.find('#deleteMember').trigger('click')

    expect($membership.deleteMembership).toHaveBeenCalled()
  })
})
