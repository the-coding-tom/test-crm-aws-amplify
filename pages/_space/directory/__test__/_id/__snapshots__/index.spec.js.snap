// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/directory/ | Directory Id Index Page renders properly 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="row py-4">
          <div class="col-md-6">
            <div class="d-flex justify-content-between">
              <h1 class="page-title"><span>Directory</span> <b>â€¢</b> Profile</h1>
              <div class="d-flex align-items-center">
                <div class="custom-control custom-switch b-custom-control-lg"><input variant="success" type="checkbox" autocomplete="off" class="custom-control-input" value="checkin" id="__BVID__8"><label class="custom-control-label" for="__BVID__8"><span class="text-muted">Checked Out</span></label></div>
              </div>
            </div>
          </div>
          <div class="col-md-6 mr-l-child-10 text-right"><a href="#_space/memberships/_id" class="btn text-primary btn-transparent" target="_self"><i class="fas fa-edit"></i> Edit Membership
            </a> <button type="button" class="btn text-primary btn-transparent"><i class="fa fa-sticky-note"></i> View Notes
            </button> <button id="deleteMember" type="button" class="btn text-danger btn-transparent"><i class="fa fa-trash"></i> Delete Member
            </button></div>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="row">
      <div class="col-md-6">
        <div class="card">
          <div class="sh-p-header"><span class="label label-default sh-label"><i class="fa fa-star"></i> Founding Member
    </span>
            <div class="d-flex align-items-center pd-20 mr-5 mr-tb-20"><img src="https://d3mym26vamg9ol.cloudfront.net/profile/1572967236_44ab6978-e4fd-4985-8ec1-0bbfc2a0f2f9." class="mr-r-20 rounded-circle avatar-xxl">
              <div>
                <h1>Ashwin Ravichandran</h1>
                <h3 class="mr-b-10">MEST</h3> <i class="ti-location-pin"></i>
                123, 456, SF, 0, India
                <br> <i class="ti-calendar"></i>
                1990-07-10
              </div>
            </div>
          </div>
          <div class="card-body">
            <div class="row">
              <div class="col-md-6"><i class="fa fa-envelope"></i> ashwin@meltwater.org
              </div>
              <div class="col-md-6"><i class="fa fa-phone"></i> +233266563950
              </div>
            </div>
            <div class="row mt-2">
              <div class="col-md-6"><i class="fa fa-venus-mars"></i> male
              </div>
              <div class="col-md-6"><i class="fa fa-globe"></i> India
              </div>
            </div>
            <div class="row mt-2">
              <div class="col-md-6"><i class="fa fa-user"></i>
              </div>
            </div>
            <div class="mt-4">
              <div class="text-muted">
                Profile
              </div>
              <p>
                The world.
              </p>
              <div>
                <div class="text-muted">
                  Cities
                </div>
                <div class="text-muted">
                  Interests
                </div> <span class="badge mr-1 badge-default">startups</span><span class="badge mr-1 badge-default">investors</span>
                <div class="text-muted mt-2">Offers</div> <span class="badge mr-1 badge-default">investments</span><span class="badge mr-1 badge-default">venture capital</span>
                <div class="text-muted mt-2">Skills</div> <span class="badge mr-1 badge-default">venture</span><span class="badge mr-1 badge-default">invest</span>
              </div>
            </div>
            <div class="mt-4">
              <p><i class="ti-id-badge"></i> 001000</p>
              <p><i class="ti-twitter-alt"></i> https://twitter.com/ashwin107</p>
              <p><i class="ti-linkedin"></i> http://www.linkedin.com/in/ashwinravichandran</p>
              <p><i class="ti-world"></i> </p>
            </div>
            <div class="sh-dls"><span> Joined on:</span> November 05, 2019
            </div>
            <div class="sh-dls"><span>Renewal due by:</span> November 06, 2020
            </div>
            <div class="sh-dls"><span>Last Checkin Timestamp:</span> January 16, 2020 02:40:55
            </div>
            <hr align="left" class="w-25 bg-dark3 text-default"> <span><div class="sh-dls"><span>Reference:</span>
          </div></span>
          <div class="sh-dls"><span> Paid by:</span>
            <nuxt-link-stub to="[object Object]">Gerhard Malah</nuxt-link-stub>
          </div>
          <div class="sh-dls"><span> Paying for:</span> </div>
        </div>
      </div>
    </div>
    <div class="col-md-6">
      <div class="card">
        <div class="card-header bg-transparent">
          <div class="d-flex justify-content-between align-items-center">
            <div class="txt-upper">
              Card Details
            </div> <button type="button" class="btn text-primary btn-transparent"><i class="fa fa-credit-card"></i> Add New Card
            </button>
          </div>
        </div>
        <div class="card-body">
          <div class="m-n25">
            <table class="table table-hover table-striped">
              <tbody>
                <tr>
                  <td>
                    MASTERCARD
                  </td>
                  <td>
                    Jun 2022
                  </td>
                  <td><button type="button" class="btn text-danger btn-transparent btn-
                        sm"><i class="fa fa-trash"></i></button></td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <!---->
        <!---->
      </div>
      <div class="card">
        <div class="card-header">
          <div class="d-flex justify-content-between align-items-center">
            <div class="txt-upper">
              MEMBERSHIP PLAN
            </div> <button id="addPlanBtn" type="button" class="btn text-primary btn-transparent"><i class="fa fa-plus"></i> Add New Plan
            </button>
          </div>
        </div>
        <div class="card-body">
          <div class="m-n25">
            <table class="table table-hover table-striped">
              <tbody>
                <tr>
                  <div>
                    <td>Individual Membership</td>
                    <td>Until 06 Nov, 2020</td>
                    <td><button id="popover-1-top" type="button" class="btn text-primary btn-transparent btn-sm"><i class="fas fa-user-shield"></i></button>
                      <b-popover-stub target="popover-1-top" triggers="hover focus" placement="top" fallbackplacement="flip" delay="50" boundary="scrollParent" boundarypadding="5" offset="0" content="Make plan primary"></b-popover-stub> <button type="button" class="btn text-primary btn-transparent btn-sm"><i class="fa fa-exchange-alt"></i></button> <button type="button" class="btn text-danger btn-transparent btn-sm"><i class="fa fa-trash"></i></button>
                    </td>
                  </div>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <!---->
        <!---->
      </div>
      <div class="card">
        <div class="card-header">
          <div class="d-flex align-items-center justify-content-between">
            <div class="txt-upper">
              CheckIn
            </div>
            <div style="width: 200px;">
              <no-ssr-stub>
                <div id="time" width="100%" input-class="form-control" lang="en" format="YYYY-MM-DD" value-type="format" confirm="" range="" type="date" placeholder="Start Date">date-picker-stub</div>
              </no-ssr-stub>
            </div>
          </div>
        </div>
        <div class="card-body">
          <div class="m-n25">
            <table role="table" aria-busy="false" aria-colcount="5" class="table b-table table-striped" id="__BVID__26">
              <!---->
              <!---->
              <thead role="rowgroup" class="">
                <!---->
                <tr role="row" class="">
                  <th role="columnheader" scope="col" aria-colindex="1" class="">Full Name</th>
                  <th role="columnheader" scope="col" aria-colindex="2" class="">Type</th>
                  <th role="columnheader" scope="col" aria-colindex="3" class="">Company</th>
                  <th role="columnheader" scope="col" aria-colindex="4" class="">Checkin Timestamp</th>
                  <th role="columnheader" scope="col" aria-colindex="5" class="">Checkout Timestamp</th>
                </tr>
              </thead>
              <tbody role="rowgroup">
                <!---->
                <tr role="row" class="b-table-empty-row">
                  <td colspan="5" role="cell" class="">
                    <div role="alert" aria-live="polite">
                      <div class="text-center my-2">There are no records to show</div>
                    </div>
                  </td>
                </tr>
                <!---->
              </tbody>
              <!---->
            </table>
          </div>
        </div>
        <!---->
        <div class="card-footer">
          <div>
            <ul class="pagination justify-content-center">
              <li class="page-item prev-page disabled"><a aria-label="Previous" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-left"></i></span></a></li>
              <li class="page-item active"><a class="page-link">1</a></li>
              <li class="page-item next-page disabled"><a aria-label="Next" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-right"></i></span></a></li>
            </ul>
          </div>
        </div>
      </div>
      <div class="card">
        <div class="card-header">
          <div class="d-flex justify-content-between">
            <div class="txt-upper">
              Events
            </div>
          </div>
        </div>
        <div class="card-body">
          <div class="m-n25">
            <div class="overflow-auto">
              <table id="my-table" role="table" aria-busy="false" aria-colcount="0" class="table b-table table-striped">
                <!---->
                <!---->
                <!---->
                <tbody role="rowgroup">
                  <!---->
                  <tr role="row" class="b-table-empty-row">
                    <td role="cell" class="">
                      <div role="alert" aria-live="polite">
                        <div class="text-center my-2">There are no records to show</div>
                      </div>
                    </td>
                  </tr>
                  <!---->
                </tbody>
                <!---->
              </table>
            </div>
          </div>
        </div>
        <!---->
        <div class="card-footer">
          <div>
            <ul role="menubar" aria-disabled="false" aria-label="Pagination" class="pagination b-pagination justify-content-center">
              <li role="presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to first page" aria-controls="my-table" aria-disabled="true" class="page-link">Â«</span></li>
              <li role="presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to previous page" aria-controls="my-table" aria-disabled="true" class="page-link">â€¹</span></li>
              <!---->
              <li role="presentation" class="page-item active"><a role="menuitemradio" aria-controls="my-table" aria-label="Go to page 1" aria-checked="true" aria-posinset="1" aria-setsize="1" tabindex="0" target="_self" href="#" class="page-link">1</a></li>
              <!---->
              <li role="presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to next page" aria-controls="my-table" aria-disabled="true" class="page-link">â€º</span></li>
              <li role="presentation" aria-hidden="true" class="page-item disabled"><span role="menuitem" aria-label="Go to last page" aria-controls="my-table" aria-disabled="true" class="page-link">Â»</span></li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
<!---->
<div id="add-plan___BV_modal_outer_" style="position: absolute; z-index: 1040;">
  <div id="add-plan" role="dialog" aria-hidden="true" aria-labelledby="add-plan___BV_modal_title_" aria-describedby="add-plan___BV_modal_body_" class="modal fade" style="display: none;">
    <div class="modal-dialog modal-md">
      <!---->
      <div role="document" id="add-plan___BV_modal_content_" tabindex="-1" class="modal-content">
        <header id="add-plan___BV_modal_header_" class="modal-header">
          <h5 id="add-plan___BV_modal_title_" class="modal-title">Add New Plan</h5><button type="button" aria-label="Close" class="close">Ã—</button>
        </header>
        <div id="add-plan___BV_modal_body_" class="modal-body">
          <div>
            <form class="">
              <div class="row">
                <fieldset id="selectPlan" class="form-group col-md-12">
                  <legend id="selectPlan__BV_label_" tabindex="-1" class="col-form-label pt-0">Plan</legend>
                  <div tabindex="-1" role="group">
                    <div class="el-select">
                      <!---->
                      <div class="el-input el-input--suffix">
                        <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Choose New Plan" class="el-input__inner">
                        <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-"></i><!----><!----><!----><!----><!----></span>
                        <!----></span>
                        <!---->
                        <!---->
                      </div>
                      <div class="el-select-dropdown el-popper" style="display: none;">
                        <div class="el-scrollbar" style="">
                          <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                            <ul class="el-scrollbar__view el-select-dropdown__list">
                              <!---->
                              <li class="el-select-dropdown__item"><span>Household Membership</span></li>
                              <li class="el-select-dropdown__item"><span>Admin Plan</span></li>
                            </ul>
                          </div>
                          <div class="el-scrollbar__bar is-horizontal">
                            <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                          </div>
                          <div class="el-scrollbar__bar is-vertical">
                            <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                          </div>
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset id="memberCard" class="form-group col-md-12">
                  <legend id="memberCard__BV_label_" tabindex="-1" class="col-form-label pt-0">Payment Source</legend>
                  <div tabindex="-1" role="group">
                    <div class="el-select">
                      <!---->
                      <div class="el-input el-input--suffix">
                        <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Choose Card" class="el-input__inner">
                        <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----><!----><!----><!----></span>
                        <!----></span>
                        <!---->
                        <!---->
                      </div>
                      <div class="el-select-dropdown el-popper" style="display: none;">
                        <div class="el-scrollbar" style="">
                          <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                            <ul class="el-scrollbar__view el-select-dropdown__list">
                              <!---->
                              <li class="el-select-dropdown__item"><span>MASTERCARD ~ **** 9399</span></li>
                            </ul>
                          </div>
                          <div class="el-scrollbar__bar is-horizontal">
                            <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                          </div>
                          <div class="el-scrollbar__bar is-vertical">
                            <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                          </div>
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <div class="row">
                <fieldset class="form-group col-md-6" id="__BVID__66">
                  <legend tabindex="-1" class="col-form-label pt-0" id="__BVID__66__BV_label_">Trial</legend>
                  <div tabindex="-1" role="group">
                    <div class="custom-control custom-checkbox"><input id="checkbox-1" type="checkbox" name="checkbox-1" autocomplete="off" class="custom-control-input" value="true"><label for="checkbox-1" class="custom-control-label">
                        Yes
                      </label></div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <!---->
              </div> <button type="submit" class="btn float-right btn-primary">Add Plan</button>
            </form>
          </div>
        </div>
        <!---->
      </div>
      <!---->
    </div>
  </div>
  <!---->
</div>
<!---->
<div role="presentation" class="el-dialog__wrapper" style="display: none;">
  <div role="document" tabindex="-1" class="el-drawer__container">
    <div aria-modal="true" aria-labelledby="el-drawer__title" role="presentation" class="el-drawer rtl" style="width: 30%;">
      <header id="el-drawer__title" class="el-drawer__header"><span role="heading">Notes</span><button aria-label="close Notes" type="button" class="el-drawer__close-btn"><i class="el-dialog__close el-icon el-icon-close"></i></button></header>
      <!---->
    </div>
  </div>
</div>
</div>
`;
